misc/core-derive/src/lib.rs

```diff
@@ -293,8 +293,8 @@ fn build_struct(ast: &DeriveInput, data_struct: &DataStruct) -> TokenStream {
         }
 
         Some(match field.ident {
-            Some(ref i) => quote!{ #elem => self.#i.inject_event(peer_id, connection_id, ev) },
-            None => quote!{ #elem => self.#field_n.inject_event(peer_id, connection_id, ev) },
+            Some(ref i) => quote!{ #elem => ::libp2p::swarm::NetworkBehaviour::inject_event(&mut self.#i, peer_id, connection_id, ev) },
+            None => quote!{ #elem => ::libp2p::swarm::NetworkBehaviour::inject_event(&mut self.#field_n, peer_id, connection_id, ev) },
```

Comment:
The idea is to keep these `::libp2p` paths on the top.

```suggestion
            Some(ref i) => quote!{ #elem => $trait_to_impl::inject_event(&mut self.#i, peer_id, connection_id, ev) },
            None => quote!{ #elem => $trait_to_impl::inject_event(&mut self.#field_n, peer_id, connection_id, ev) },
```